website/docs/getting-started/setup/instance-configuration/file-size-limit.md

```diff
@@ -0,0 +1,47 @@
+# Upload File Size Limit
+
+In self-hosted Appsmith instances, the default file size limit is 150 MB. You can adjust the limit to support larger files. This page explains how to update the file size limit and configure reverse proxies or load balancers, if used, to ensure the new limits are applied.
+
+## Change file size limit
+
+To accommodate large files, update the Appsmith instance configuration by setting the `APPSMITH_CODEC_SIZE` environment variable in your deployment configuration. When installed on:
+
+- **Docker**: Update the `docker.env` file.
+- **Kubernetes**: Update the `values.yaml` file.
+
+For example, to increase the file size limit to 500 MB, use the following configuration:
+
+```bash
+APPSMITH_CODEC_SIZE=500MB
+```
+
+## Change reverse proxy or load balancer limit
+
+If your Appsmith instance is behind a reverse proxy or load balancer, you also need to configure the file size limits in the proxy or load balancer settings. Below are example configurations:
+
+* **Nginx**: Add or update the following directive in your Nginx configuration file. For example, to set a size limit of 500 MB, update the configuration as below:
+
+ ```bash
+  client_max_body_size 500M;
+ ```
+
+* **Caddy**: Set the `max_request_body` directive in your Caddy configuration file. For example, to set a size limit of 500 MB, update the configuration as below:
+
+ ```bash
+  max_request_body 500M;
+ ```
+
+* **Load balancers**: Configure the `proxy-body-size` or similar directive in your load balancer settings. For example, to set a size limit of 500 MB, update the configuration as below:
+
+ ```bash
+  proxy-body-size 500M;
+ ```
+
+## Important considerations
+
+- Ensure you have configured all relevant components: Appsmith instance, reverse proxy, and load balancer to handle the desired file size limits. 
+- If you’re using the FilePicker widget, make sure that the limits set for the [Max number of files](/reference/widgets/filepicker#max-no-of-filesnumber) and [Max file size](/reference/widgets/filepicker#max-file-sizenumber) properties fall within the file size limits configured for your Appsmith instance.
```

Comment:
@rahulbarwal - Could you please review this point and confirm if it makes sense?

We added this point because we believe the issue may stem from the following:

- 	The instance-level file size limit is set to 4 MB.
- 	The maximum number of files allowed is 5, with a maximum file size of 1 MB each.

If a user uploads 5 files in a single request, the total file size will exceed the 4 MB instance-level limit, which should cause the upload to fail. Is this the expected behavior and is the information provided correct?

Reply:
Makes sense.